base/src/main/java/org/aya/concrete/resolve/visitor/StmtResolver.java

```diff
@@ -32,9 +32,9 @@ static void resolveStmt(@NotNull Stmt stmt, @NotNull ResolveInfo info) {
     switch (stmt) {
       case Command.Module mod -> resolveStmt(mod.contents(), info);
       case Decl.DataDecl decl -> {
-        var signatureResolver = new ExprResolver(new ExprResolver.Options(true));
+        var signatureResolver = new ExprResolver(ExprResolver.LAX);
         var local = signatureResolver.resolveParams(decl.telescope, decl.ctx);
-        decl.telescope = local._1;
+        decl.telescope = local._1.prependedAll(signatureResolver.allowedGeneralizes().valuesView());
```

Comment:
I guess we can move `.prependedAll(signatureResolver.allowedGeneralizes().valuesView())` to `signatureResolver.resolveParams`

Reply:
Related #207

Reply:
> I guess we can move `.prependedAll(signatureResolver.allowedGeneralizes().valuesView())` to `signatureResolver.resolveParams` 

Are you gonna do it?

Reply:
Maybe tomorrow

Reply:
You're so fast. I just pulled the code and realized that you have done it. 🤣